snuba/clickhouse/translators/snuba/mappers.py

```diff
@@ -44,14 +47,77 @@ def attempt_map(
             and expression.table_name == self.from_table_name
         ):
             return ColumnExpr(
-                alias=expression.alias,
+                alias=expression.alias
+                or qualified_column(self.from_col_name, self.from_table_name or ""),
                 table_name=self.to_table_name,
                 column_name=self.to_col_name,
             )
         else:
             return None
 
 
+@dataclass(frozen=True)
+class ColumnToLiteralMapper(ColumnMapper):
+    """
+    Maps a column name into a hardcoded literal in the query preserving
+    the alias.
+    """
+
+    from_table_name: Optional[str]
+    from_col_name: str
+    to_literal_value: OptionalScalarType
+
+    def attempt_map(
+        self,
+        expression: ColumnExpr,
+        children_translator: SnubaClickhouseStrictTranslator,
+    ) -> Optional[LiteralExpr]:
+        if (
+            expression.table_name == self.from_table_name
+            and expression.column_name == self.from_col_name
+        ):
+            return LiteralExpr(
+                alias=expression.alias
+                or qualified_column(self.from_col_name, self.from_table_name or ""),
```

Comment:
I think it is appropriate to use `or` innstead of `is not None else` here because this is a string and the behavior of a string in conditions like `if not mystr:` is well known.